### ISSUE TITLE
phpoffice/phpexcel is abandoned

### ISSUE THREAD

**Issue by Traumflug** [thirtybees/thirtybees#492]
Composer kindly shows it:

```
$ composer update --no-dev
[...]
Package phpoffice/phpexcel is abandoned, you should avoid using it. Use phpoffice/phpspreadsheet instead.
```

**Comment by Traumflug** [comment]
The unfortunate thing is, *phpoffice/phpspreadsheet* requires PHP 5.6, while thirty bees should work with PHP 5.5 as well. See https://phpspreadsheet.readthedocs.io/en/develop/

It looks like this upgrade has to wait.

**Comment by firstred** [comment]
Yeah same feeling here. We'll have to make do with this deprecated version.

**Comment by getdatakick** [comment]
Proposed solution: 

- Refactor import functionality to use generic data source (not csv)
- add ability to register data source readers by modules
- implement default csv data source reader in core
- introduce new module to support import from excel, based on `phpoffice/phpspreadsheet` (or any other library)
- remove `phpoffice/phpexcel`  dependency from core

This way, we can get remove unnecessary library dependency from core, and still support import from excel files

**Comment by getdatakick** [comment]
Closed by commit f0e43e58651941353f10147aac1a4cdea65b3b66
